whisper/decoding.py

```diff
@@ -721,8 +725,7 @@ def run(self, mel: Tensor) -> List[DecodingResult]:
                 )
             ]
 
-        # repeat the audio & text tensors by the group size, for beam search or best-of-n sampling
-        audio_features = audio_features.repeat_interleave(self.n_group, dim=0)
```

Comment:
@wangchou  I was wondering if you remove the repeat of audio_features, where you repeat the kv_cache for cross attention? Otherwise, during cross_attention, q@k seems with mismatch dims since tokens are repeated according the beam_size.

Reply:
@jongwook, would you mind checking this please? Thanks.

Reply:
@yuekaizhang @ operator (tf.matmul) supports broadcasting since [v1.12.1](https://github.com/tensorflow/tensorflow/commit/47ab68d265a96b6e7be06afd1b4b47e0114c0ee9).

Reply:
```
import torch
q = torch.ones(70, 4, 16, 4)
k = torch.ones(7, 4, 4, 16)
k2 = torch.ones(70, 4, 4, 16)

context2 = q @ k2
print(context2.shape)

context1 = q @ k
print(context1.shape)
```
I ran with torch==2.0.1
```
RuntimeError: The size of tensor a (70) must match the size of tensor b (7) at non-singleton dimension 0
```

Reply:
@yuekaizhang I ran whisper with beam_size=5. It works.
```
python -m whisper ../samples/thatBand2ch_short.wav --language ja --model small --beam_size=5
```

After adding print in qkv_attention() like
```python
        qk = q @ k
        print("q.shape=",q.shape,", k.shape=", k.shape)
```

it outputs
```
...
q.shape= torch.Size([5, 12, 1, 64]) , k.shape= torch.Size([5, 12, 64, 6])
q.shape= torch.Size([5, 12, 1, 64]) , k.shape= torch.Size([1, 12, 64, 1500])
```

What arguments did you use to get k like (7, 4, 4, 16)? Where is that **7** comes from?

Reply:
@wangchou  Did you try inference with batch_size > 1? I met this issue when I tried with both batch_size, beam_size > 1.

Reply:
@yuekaizhang I don't even know ```batch_size``` option. And I cannot find it with ```whisper --help```. Sorry.